### PR TITLE
Subs.php - Faster/Better Parser Modification

### PR DESCRIPTION
**Pull request by ghost**
Please see http://www.simplemachines.org/community/index.php?topic=538611.0 for a description of the issue and my solution.  Thank you!

Signed-off-by: Douglas Orend douglas.orend@comcast.net

### CODE REVIEW

File: Sources/Subs.php [hunk 1]
```diff
@@ -899,6 +899,28 @@ function permute($array)
 	return $orders;
 }
 
+// Rearranges all parameters to be in the right order.  Returns TRUE if no parameters are leftover.
+function_param_order($message, &$parameters, &$out)
```
Comment (live627) [inline Sources/Subs.php:903]: typo

File: Sources/Subs.php [hunk 2]
```diff
@@ -2101,23 +2123,16 @@ function parse_bbc($message, $smileys = true, $cache_id = '', $parse_tags = arra
 			// This is long, but it makes things much easier and cleaner.
 			if (!empty($possible['parameters']))
 			{
-				$preg = array();
+				// START: Dougiefresh's faster parse_bbc function
```
Comment (live627) [inline Sources/Subs.php:2126]: ???

File: Sources/Subs.php (hunk 1 above)
Comment (ghost) [inline Sources/Subs.php:903]: Whoops!  I meant to use the following:
function fix_param_order($message, &$parameters, &$out)

File: Sources/Subs.php (hunk 2 above)
Comment (ghost) [inline Sources/Subs.php:2126]: What's the "???" for?  Just wondering!  Thanks!

Comment (XinYenFon) [inline Sources/Subs.php:2126]: ??? = Why you left your personal mark there

Comment (ghost) [inline Sources/Subs.php:2126]: I don't know.  Remove it if you want.....  If I need to do another commit without "my personal mark", let me know....

Comment (Oldiesmann) [inline Sources/Subs.php:2126]: We can't remove it unless it's merged into SMF, so you'll need to submit another commit to remove it.

File: Sources/Subs.php [hunk 3]
```diff
@@ -899,6 +899,28 @@ function permute($array)
 	return $orders;
 }
 
+// Rearranges all parameters to be in the right order.  Returns TRUE if no parameters are leftover.
+function_param_order($message, &$parameters, &$out)
+{
+	$params = explode(' ', substr($message, 0, strpos($message, ']')));
```
Comment (live627) [inline Sources/Subs.php:905]: Seems to me that this will break horribly with loose brackets.

Comment (ghost) [inline Sources/Subs.php:905]: Please provide an example.

Comment (live627) [inline Sources/Subs.php:905]:
```
[quote author=[Unknown] link=topic=353126.msg2523882#msg2523882 date=1267759657]
text

-[Unknown]
[/quote]
```

Comment (ghost) [inline Sources/Subs.php:905]: If I were to check for the existance of an opening bracket, and scan for the closing bracket, then look for yet another closing bracket, do you think that would solve this problem?
